src/coreclr/src/jit/gentree.cpp

```diff
@@ -3252,17 +3252,44 @@ unsigned Compiler::gtSetEvalOrder(GenTree* tree)
         switch (oper)
         {
 #ifdef TARGET_ARM
-            case GT_CNS_LNG:
-                costSz = 9;
-                costEx = 4;
-                goto COMMON_CNS;
-
             case GT_CNS_STR:
                 // Uses movw/movt
                 costSz = 7;
                 costEx = 3;
                 goto COMMON_CNS;
 
+            case GT_CNS_LNG:
```

Comment:
Wondering why this is needed since the PR is for Arm64? Isn't `GT_CNS_LNG` never created for 64-bit JIT?

Reply:
Sorry, I worded this poorly. I meant to say the PR title is implying this is just Arm64 work and should likely be updated if its also touching arm32.

Reply:
It current;y enables CSE's of constants across all platforms.
On ARM64 we get the largest benefits..